### PR TITLE
refactor(ngSanitize) Remove workarounds for IE8

### PR DESCRIPTION
**Pull request by realityking**
Also provides better tests.

### CODE REVIEW

**Comment by Narretz** [comment]
What do you mean by "better tests"? I assume you removed those tests that were testing the IE8 specific functions?

**Comment by realityking** [comment]
@Narretz Yes, I removed those testing IE8 specific workarounds and added some tests to tests the general functionality - otherwise there would have been no tests left.

**Comment by Narretz** [comment]
Okay, then you should split this into commits - the first includes the new tests, the second removes all IE8 stuff.

**Comment by realityking** [comment]
@Narretz done

**Comment by realityking** [comment]
@Narretz The only failed job seems to be related to saucelabs. Can this be merged?

**Comment by Narretz** [comment]
Hi, sorry for the late response. I have another very small nitpick: our commit messages have a `:` after the `refactor(ngSanitize)` part, can you fix that up?

@petebacondarwin are you okay with removing this IE8 workaround?

**Comment by realityking** [comment]
@Narretz Heh, never noticed that. Fixed now.

**Comment by petebacondarwin** [comment]
@realityking - more commit message nitpicking :-) We don't capitalize the first letter of the description.

That being said, I am happy to help strip out unnecessary code from the codebase and get this in.
Thanks!

File: test/ngSanitize/sanitizeSpec.js [hunk 1]
```diff
@@ -511,37 +510,13 @@ describe('decodeEntities', function() {
     module('ngSanitize');
   });
 
-  afterEach(function() {
-    window.hiddenPre = origHiddenPre;
+  it('should unescape text', function() {
+    htmlParser('a&lt;div&gt;&amp;&lt;/div&gt;c', handler)
```
Comment (petebacondarwin) [inline test/ngSanitize/sanitizeSpec.js]: missing semi-colon

File: test/ngSanitize/sanitizeSpec.js [hunk 2]
```diff
@@ -511,37 +510,13 @@ describe('decodeEntities', function() {
...
-    window.hiddenPre = {
-      textContent: '',
-      innerText: 'INNER_TEXT'
-    };
-    inject(function($sanitize) {
-      htmlParser('<tag>text</tag>', handler);
-      expect(text).toEqual('');
-    });
+  it('should preserve whitespace', function() {
+    htmlParser('  a&amp;b ', handler)
```
Comment (petebacondarwin) [inline test/ngSanitize/sanitizeSpec.js]: missing semi-colon

**Comment by petebacondarwin** [comment]
It is always good to run `grunt test` on our PRs before submitting. Apart from building, minifying and running all the unit and e2e tests locally, it also runs `grunt ci-checks` which checks the jshint and jscs rules.

**Comment by realityking** [comment]
@petebacondarwin I always run `grunt test` but I hadn't yet discovered `grunt ci-checks`, thanks for the pointer.

**Pull request closed without merging**